libs/api/domains/health-directorate/src/lib/health-directorate.service.ts

```diff
@@ -123,6 +134,119 @@ export class HealthDirectorateService {
           ),
         }
       }) ?? []
+
     return { vaccinations }
   }
+
+  /* Waitlists */
+  async getWaitlists(auth: Auth, locale: Locale): Promise<Waitlists | null> {
+    const data = await this.healthApi.getWaitlists(auth, locale)
+
+    if (!data) {
+      return null
+    }
+    const waitlists: Array<Waitlist> =
+      data.map((item) => {
+        return {
+          id: item.id,
+          lastUpdated: item.lastUpdated
+            ? new Date(item.lastUpdated?.toString())
```

Comment:
Might want to do a `NaN` check when converting a string to a date

Reply:
@coderabbitai can you write a safe date conversion function, that checks if date is nullable, or if the converted date is `NaN`